src/rgw/rgw_admin.cc

```diff
@@ -1975,7 +1975,7 @@ static void sync_status(Formatter *formatter)
 
   list<string> md_status;
 
-  if (zone.id == zonegroup.master_zone) {
+  if (zonegroup.is_master && zone.id == zonegroup.master_zone) {
```

Comment:
looks right. there's a `store->is_meta_master()` function that does the same thing - could you use that instead?